packages/ember-data/lib/system/relationships/has_many.js

```diff
@@ -104,10 +104,14 @@ function hasMany(type, options) {
     key: null
   };
 
-  return Ember.computed(function(key) {
+  return Ember.computed(function(key, records) {
     var relationship = this._relationships[key];
+    if (arguments.length === 2) {
+      relationship.removeRecords(relationship.getRecords());
```

Comment:
nice catch you don't want to do a `getRecords()` here, there is a relationship.clear() you can use instead